core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -366,6 +370,9 @@ public AllocationStatus getLastAllocationStatus() {
      */
     public long getRemainingDelay(final long nanoTimeNow, final Settings indexSettings) {
         long delayTimeoutNanos = INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.get(indexSettings).nanos();
+        if (delayTimeoutNanos < 0) {
```

Comment:
instead of baking it here, I would put in the setting parser. This means you'll catch all the places where the setting is used in one place.